### ISSUE TITLE
Infrared image colorization

### ISSUE THREAD

**Issue by songyn95** [msracver/Deep-Exemplar-based-Colorization#41]
- At present, the test gray-scale image colorization effect is very good, but when I use infrared image colorization, it has no effect. How can I improve it?

**Comment by hmmlillian** [comment]
Hi @songyn95 , since our model is trained with grayscale images in ImageNet dataset, I do not think it can be directly applied to infrared images. You may need to re-train or re-design the network for the task of infrared image colorization using infrared images.

**Comment by songyn95** [comment]
@hmmlillian  Which network do you think needs to be retrained? vgg19_ Grey, colornet or something else? At present, I may feel that I need to retrain the vgg19 network with infrared images.
It is worth mentioning that is there a good example of training vgg19 network? I want to refer to it

**Comment by hmmlillian** [comment]
@songyn95 I think both networks (vgg19_gray and colorization subnet) need to be retrained, since the input image of both networks have to be replaced by infrared images. To train a classification network like vgg19, you need the class labels of infrared images and refer to the original paper of vgg network.

**Comment by songyn95** [comment]
1. Since I do image translation, that is, infrared image to RGB image, the number of categories cannot be distinguished. How to solve this situation?
2. Can I provide the training script of colorization subnet so that I can understand how to train

**Comment by songyn95** [comment]
In addition, I have another question. If I set require_grad=True when loading pretraining model. Can we achieve better results without retraining（infrared image to RGB image）？Examples are as follows：
```
color_net = ExampleColorNet(ic=13)
color_net.load_state_dict(torch.load(opt.train_model, map_location=lambda storage, loc: storage))
for param in color_net.parameters():
        param.requires_grad = True
```
